core/src/matrix.rs

```diff
@@ -85,3 +86,623 @@ impl std::ops::MulAssign for Matrix {
         }
     }
 }
+
+impl AbsDiffEq for Matrix {
+    type Epsilon = <f32 as AbsDiffEq>::Epsilon;
+    fn default_epsilon() -> Self::Epsilon {
+        f32::default_epsilon()
+    }
+
+    fn abs_diff_eq(&self, other: &Self, epsilon: Self::Epsilon) -> bool {
+        (self.a.abs_diff_eq(&other.a, epsilon) || (self.a.is_nan() && self.a.is_nan()))
+            && (self.b.abs_diff_eq(&other.b, epsilon) || (self.b.is_nan() && self.b.is_nan()))
+            && (self.c.abs_diff_eq(&other.c, epsilon) || (self.c.is_nan() && self.c.is_nan()))
+            && (self.d.abs_diff_eq(&other.d, epsilon) || (self.d.is_nan() && self.d.is_nan()))
+            && (self.tx.abs_diff_eq(&other.tx, epsilon) || (self.tx.is_nan() && self.tx.is_nan()))
+            && (self.ty.abs_diff_eq(&other.ty, epsilon) || (self.ty.is_nan() && self.ty.is_nan()))
```

Comment:
I'm not sure if the `NaN` checks should be in here -- looking at the [approx crate impl for `f64`](https://docs.rs/approx/0.3.2/src/approx/abs_diff_eq.rs.html#75), they don't do `NaN` checks there, so we should follow suit. If we need to check against NaN specifically, we should explicitly check for it (possibly with a separate function).

Additionally, I think the ActionScript `Matrix` will be it's own separate object up in AVM land, comapred to this `Matrix` which is used internally by the renderer.  Specifically the ActionScript `Matrix` uses floats for all the values, while this Matrix has to keep the translation in twips (1 twip or 0.05 pixel is the minimum change in position of a Flash display object). So I think these renderer matrices should always be valid invertible, and the `NaN` tests should be unnecessary.

Maybe best would be to make `invert` return an `Option` with `None` if it's non-invertible.